application/controllers/Lotw.php

```diff
@@ -524,10 +524,11 @@ private function loadFromFile($filepath, $display_view = "TRUE")
 						$state = "";
 					}
 					// Present only if the QSLing station specified a single valid grid square value in its station location uploaded to LoTW.
+					$qsl_gridsquare = "";
 					if (isset($record['gridsquare'])) {
-						$qsl_gridsquare = $record['gridsquare'];
-					} else {
-						$qsl_gridsquare = "";
+						if (strlen($record['gridsquare']) > strlen($status[2]) || substr(strtoupper($status[2]), 0, 4) != substr(strtoupper($record['gridsquare']), 0, 4)) {
```

Comment:
Sri. but this would mean: We take OUR grid (from WL) if we're more precise than LotW.

e.g.: LotW cnfms with JO30, Op entered JO31oo. --> JO31oo is taken because it's more precise
that's wrong imho

Reply:
> e.g.: LotW cnfms with JO30, Op entered JO31oo. --> JO31oo is taken because it's more precise that's wrong imho

Why is that wrong? For VUCC / LoTW only 4-char grids count AFAIK. So why not keep our 2-char more specific suffix?

Reply:
> Why is that wrong? For VUCC / LoTW only 4-char grids count AFAIK. So why not keep our 2-char more specific suffix?

Just disregard my comment... too blind to read.

Reply:
> Why is that wrong? For VUCC / LoTW only 4-char grids count AFAIK. So why not keep our 2-char more specific suffix?

Different 4-char grid is covered by the 2nd part or the OR stuff.

Reply:
Sorry - reversed boolean logic is confusing sometimes.
This one works correct.